src/Service/AdminOrderService.php

```diff
@@ -302,23 +310,44 @@ public function refreshLabels(array $labelIds): array
         $orderLabels = [];
 
         foreach ($collection as $consignment) {
-            $orderLabel         = OrderLabel::findByLabelId($consignment->getConsignmentId());
-            $orderLabel->status = MyParcelStatusProvider::getInstance()
-                ->getStatus($consignment->getStatus());
-            $orderLabel->save();
-            $orderLabels[] = $orderLabel;
-
-            OrderLogger::addLog(
-                [
-                    'order'   => $orderLabel->id_order,
-                    'message' => "Refreshed label $orderLabel->id_label",
-                ]
-            );
+            $orderLabels[] = $this->consignmentToOrderLabel($consignment);
         }
 
         return $orderLabels;
     }
 
+    /**
+     * @param  object $consignment
+     *
+     * @return object $orderLabel
+     * @throws \PrestaShopException
+     */
+    private function consignmentToOrderLabel(object $consignment): object
```

Comment:
object??? dit kan niet in php 7.1, zorg dat je editor goed staat. Plus dit moet geen object zijn maar gewoon een AbstractConsignment